src/main/scala/epic/parser/models/PositionalNeuralModelFactory.scala

```diff
@@ -0,0 +1,357 @@
+import epic.features.SurfaceFeaturizer.SingleWordSpanFeaturizer
+import epic.features._
+import epic.framework.Feature
+import epic.lexicon.Lexicon
+import epic.parser.projections.GrammarRefinements
+import epic.trees._
+import epic.trees.annotations.TreeAnnotator
+import epic.util.{LRUCache, Optional}
+import epic.dense.Transform
+import epic.dense.TanhTransform
+import epic.dense.OutputTransform
+import epic.dense.AffineOutputTransform
+import epic.dense.OutputEmbeddingTransform
+import epic.dense.Word2Vec
+import scala.collection.mutable.HashMap
+import epic.dense.Word2VecSurfaceFeaturizerIndexed
+import epic.dense.Word2VecDepFeaturizerIndexed
+import epic.dense.Word2VecIndexed
+import epic.dense.FrequencyTagger
+import epic.dense.CachingLookupTransform
+import epic.dense.CachingLookupAndAffineTransformDense
+import epic.dense.EmbeddingsTransform
+import epic.dense.NonlinearTransform
+import scala.io.Source
+import scala.collection.mutable.HashSet
+import epic.dense.BatchNormalizationTransform
+
+/**
+ * Entry point for instantiating a neural CRF parser. Parameters specify neural
+ * net parameters, word vectors, and sparse features to use.
+ *
+ * @author gdurrett
+ **/
+
+
+/**
+ * Less-used parameters
+ */
+case class ExtraPNMParams(@Help(text="Used for ablations with random word embeddings; don't change this. Options: normal, random, trivial, normalpos")
+                          embeddingType: String = "normal",
+                          @Help(text="Use longest frequent suffix (standard representation) for sparse feats")
+                          useSparseLfsuf: Boolean = true,
+                          @Help(text="Use sparse Brown cluster features")
+                          useSparseBrown: Boolean = false,
+                          @Help(text="Use expanded set of sparse surface features (doesn't help)")
+                          useMostSparseIndicators: Boolean = false,
+                          @Help(text="Scaling factor for all input vectors")
+                          vectorRescaling: Double = 1.0,
+                          @Help(text="Use the output embedding model (Figure 4b in the neural CRF paper)")
+                          outputEmbedding: Boolean = false,
+                          @Help(text="Dimension of the output embedding model")
+                          outputEmbeddingDim: Int = 20,
+                          @Help(text="When initializing the output embedding model, initialize based on root symbols")
+                          coarsenByRoot: Boolean = false,
+                          @Help(text="Use separate neural net parameters for span/unary/binary settings. Doesn't help.")
+                          decoupleTransforms: Boolean = false,
+                          @Help(text="Extract additional output features based on root label.")
+                          useRootLabel: Boolean = false,
+                          @Help(text="Set unknown word vectors to be random rather than 0")
```

Comment:
Nice idea, though I found that what I did with unks basically didn't matter. Don't remember if it was different in sparse+dense vs. dense, but my hypothesis is that unks probably often easy things from the parser's point of view (part of named entities, etc.).